### PR TITLE
DNA: Compat Package

### PR DESCRIPTION
**Pull request by gravityrail**
This PR adds a backwards compatibility package.
Intended to be the place to add deprecated classes, methods, and global functions in order to smooth the deprecation process.

#### Changes proposed in this Pull Request:
* Add a Compat Package

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Part of the Jetpack DNA project - p1HpG7-70O-p2

#### Testing instructions:
* Verify tests still pass

#### Proposed changelog entry for your changes:
* Add Jetpack Backwards compatibility package

### CODE REVIEW

File: class.jetpack-client.php [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?php
+/**
+ * Deprecated since 7.5 in favor of packages/connection/src/Client.php
```
Comment (dereksmart) [inline class.jetpack-client.php]: https://developer.wordpress.org/reference/functions/_deprecated_file/

File: packages/compat/legacy/class.jetpack-client.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * @deprecated Use Automattic\Jetpack\Connection\Client
+ */
+class Jetpack_Client {
+
+	/**
+	 * @deprecated use Automattic\Jetpack\Connection\Client::remote_request
+	 */
+	static function remote_request( $args, $body = null ) {
+		return Client::remote_request( $args, $body );
```
Comment (dereksmart) [inline packages/compat/legacy/class.jetpack-client.php]: Let's throw one of these https://developer.wordpress.org/reference/functions/_deprecated_function/

**Comment by dereksmart** [comment]
Pushed a commit that uses
`_deprecated_function()`
and 
`_deprecated_file()`

I used this snippet to test in a functionality plugin, and verified the notices are thrown and that there is no fatal. 

```<?php
/*
Plugin Name: Test Back Compat
Version: 1.0
*/

add_action( 'plugins_loaded', function() {
	// Should not fatal
	require_once JETPACK__PLUGIN_DIR . '/class.jetpack-client.php';
	var_dump( Jetpack_Client::remote_request( array(), null ), 1 );
});
```

**Comment by dereksmart** [comment]
I am going to merge this to the feature/back-compat-7.5 branch now.

**Comment by gravityrail** [comment]
Thank you!

**Pull request merged**